nabs/preprocessors.py

```diff
@@ -0,0 +1,291 @@
+        This will be a callable generator function.
+
+    See Also
+    --------
+    :func:`daq_step_scan_wrapper`
+    :func:`daq_step_scan_standard_args`
+    """
+
+    @wraps(plan)
+    def inner(*args, **kwargs):
+        events = kwargs.pop('events', None)
+        duration = kwargs.pop('duration', None)
+        record = kwargs.pop('record', True)
+        use_l3t = kwargs.pop('use_l3t', False)
+        return (yield from daq_step_scan_wrapper(plan(*args, **kwargs),
+                                                 events=events,
+                                                 duration=duration,
+                                                 record=record,
+                                                 use_l3t=use_l3t))
+    return inner
+
+
+def daq_step_scan_standard_args(events=None, duration=None, record=True,
+                                use_l3t=False):
+    """
+    No-op function to hold template parameter info for generated docs.
+
+    Parameters
+    ----------
+    events : int, optional
+        Number of events to take at each step. If omitted, uses the
+        duration argument or the last configured value.
+
+    duration : int or float, optional
+        Duration of time to spend at each step. If omitted, uses the events
+        argument or the last configured value.
+
+    record : bool, optional
+        Whether or not to record the run in the DAQ. Defaults to True because
+        we don't want to accidentally skip recording good runs.
+
+    use_l3t : bool, optional
+        Whether or not the use the l3t filter for the events argument. Defaults
+        to False to avoid confusion from unconfigured filters.
+    """
+
+    pass
+
+
+def daq_during_wrapper(plan, record=True, use_l3t=False, controls=None):
+    """
+    Wrap a plan so that the DAQ runs at the same time.
+
+    This can be used with an ordinary bluesky plan that you'd like the daq
+    to run along with. This also stages the DAQ so that the run start/stop
+    will be synchronized with the bluesky runs.
+
+    Note that this is not a calib cycle scan. See
+    :func:`daq_step_scan_wrapper` and :func:`daq_step_scan_decorator`
+    for the calib cycle variant.
+
+    All configuration must be done by supplying config kwargs to this wrapper.
+
+    This must be applied outside the run_wrapper.
+
+    The :func:`daq_during_decorator` is the same as the
+    :func:`daq_during_wrapper`, but it is meant to be used as a function
+    decorator.
+
+    Parameters
+    ----------
+    plan : plan
+        A bluesky plan that yields bluesky Msg objects.
+
+    record : bool, optional
+        Whether or not to record the run in the DAQ. Defaults to True because
+        we don't want to accidentally skip recording good runs.
+
+    use_l3t : bool, optional
+        Whether or not the use the l3t filter for the events argument. Defaults
+        to False to avoid confusion from unconfigured filters.
+
+    controls : list of readables, optional
```

Comment:
Does the "readable" API include `.position`/`.get()`? (Comment above talks about this same topic more, though this borders on being nitpicky)

Reply:
Ah! It does not. "readable" includes `.read()`, "settable" includes `position` as an optional. Neither includes `.get()`. The type here must be changed to clarify what is currently required.